src/shared/utils/windowPeriod/fromTimeRange.ts

```diff
@@ -0,0 +1,44 @@
+import {TimeRange} from 'src/types'
+import {
+  millisecondsToDuration,
+  durationToMilliseconds,
+  parseDuration,
+} from 'src/shared/utils/duration'
+import {convertMillisecondDurationToWindowPeriod} from 'src/shared/utils/windowPeriod'
+
+/**
+ * @param timeRange -- timeRange, UI type
+ * @returns {number} -- milliseconds of windowPeriod
+ */
+const windowPeriodFromTimeRange = (timeRange: TimeRange): number => {
+  switch (timeRange.type) {
+    case 'selectable-duration':
+      return timeRange.windowPeriod
```

Comment:
Yes. But it can still be overwritten by the user.

* Scenario 1:  user doesn't provide their own v.windowPeriod. So use the windowPeriod for that timeRange. In this case, `SELECTABLE_TIME_RANGES` says it's 10000.
   ```
   sample.data(set: "airSensor")
        |> range(start: -5m, stop: now())
   ```
* Scenario 2:  user overwrites the v.windowPeriod. Take the user's input, and set to 2444.
   ```
   v = {windowPeriod: 2444}
   sample.data(set: "airSensor")
        |> range(start: -5m, stop: now())
   ```